core/src/main/java/org/apache/accumulo/core/fate/FateKey.java

```diff
@@ -168,4 +168,12 @@ private static Optional<ExternalCompactionId> deserializeCompactionId(FateKeyTyp
         throw new IllegalStateException("Unexpected FateInstanceType found " + type);
     }
   }
+
+  @Override
+  public String toString() {
+    return "[" + getClass().getSimpleName() + " FateKeyType:" + type
+        + (keyExtent.isPresent() ? ", KeyExtent:" + keyExtent.orElseThrow()
+            : ", ExternalCompactionID:" + compactionId.orElseThrow())
+        + "]";
+  }
```

Comment:
Could use ToStringBuilder here. It makes it so we don't have to deal with formatting as much (like placement of spaces, brackets or commas).
```suggestion
  @Override
  public String toString() {
    var buf = new ToStringBuilder(this, ToStringStyle.SHORT_PREFIX_STYLE);
    buf.append("FateKeyType", type);
    keyExtent.ifPresentOrElse(keyExtent -> buf.append("KeyExtent", keyExtent),
        () -> buf.append("ExternalCompactionID", compactionId.orElse(null)));
    return buf.toString();
  }
```

This does slightly change the output but I think it looks fine. Here is the new output:
```
FateKey[FateKeyType=COMPACTION_COMMIT,ExternalCompactionID=ECID-1918a419-f4d1-4c91-bb1e-c9190ec56e55]
FateKey[FateKeyType=SPLIT,KeyExtent=MetaFateStoreFateIT_testCreateWithKeyInProgress0;zzz;aaa]
```
and here is the original:
```
[FateKey FateKeyType:COMPACTION_COMMIT, ExternalCompactionID:ECID-537f9fe4-4220-4238-9a2c-bc20b0dd4c59]
[FateKey FateKeyType:SPLIT, KeyExtent:MetaFateStoreFateIT_testCreateWithKeyInProgress0;zzz;aaa]
```

Reply:
Thanks this is nicer
3ca39d0d57fbf5321b3ca5abc8fee1b16a52c37f addresses this.
Only thing different is 
`compactionId.orElse(null)`
instead I have
`compactionId.orElseThrow()`
since either one of KeyExtent or ExternalCompactionID should be present. This is handled in the FateKey object creation though and maybe shouldn't be added handling in a toString()... shouldn't make a difference either way though.